docs/en/modules/coap.md

```diff
@@ -6,225 +6,148 @@
 
-## coap.Server()
 
-Creates a CoAP server.
+## coap.new()
+
+Creates a CoAP instance.
 
 #### Syntax
-`coap.Server()`
+`coap.new()`
 
 #### Parameters
 none
 
 #### Returns
-CoAP server
+CoAP peer instance
 
 #### Example
 ```lua
--- use copper addon for firefox
-cs=coap.Server()
-cs:listen(5683)
-
-myvar=1
-cs:var("myvar") -- get coap://192.168.18.103:5683/v1/v/myvar will return the value of myvar: 1
-
-all='[1,2,3]'
-cs:var("all", coap.JSON) -- sets content type to json
-
--- function should tack one string, return one string.
-function myfun(payload)
-  print("myfun called")
-  respond = "hello"
-  return respond
-end
-cs:func("myfun") -- post coap://192.168.18.103:5683/v1/f/myfun will call myfun
+-- Create UDP Socket to send or receive data
+u = net.createUDPSocket()
+-- Create CoAP instance to handle data;
+c = coap.new()
+u:listen(5683)
+-- The function is called when data is received
+u:on("receive", function(s, data, port, ip) c:receive(data,ip,port) end)
+-- Set CoAP sender
+c:sender(function(data,ip,port) u:send(port,ip,data) end)
 
 ```
 
-# CoAP Client
 
-## coap.client:get()
+## coap.peer:handler()
 
-Issues a GET request to the server.
+Set the CoAP server request handler.
 
 #### Syntax
-`coap.client:get(type, uri[, payload])`
+`coap.peer:handler(function(pkt,ip,port))`
 
 #### Parameters
-- `type` `coap.CON`, `coap.NON`, defaults to CON. If the type is CON and request fails, the library retries four more times before giving up.
-- `uri` the URI such as "coap://192.168.18.103:5683/v1/v/myvar", only IP addresses are supported i.e. no hostname resoltion.
-- `payload` optional, the payload will be put in the payload section of the request.
-
-#### Returns
-`nil`
+- `function(data,ip,port)`callback function.
 
-## coap.client:put()
+The first parameter of callback is the `CoAP package` instance that receive from remote.
+The second and third parameter of callback is the address of the data being received.
 
-Issues a PUT request to the server.
-
-#### Syntax
-`coap.client:put(type, uri[, payload])`
-
-#### Parameters
-- `type` `coap.CON`, `coap.NON`, defaults to CON. If the type is CON and request fails, the library retries four more times before giving up.
-- `uri` the URI such as "coap://192.168.18.103:5683/v1/v/myvar", only IP addresses are supported i.e. no hostname resoltion.
-- `payload` optional, the payload will be put in the payload section of the request.
+You must return the first argument after processing the request.
```

Comment:
I don't understand this statement.... Are you saying that the callback function must return the first argument? Why?